Bottom: 4564de30978967554ad3783475722a811ccea2e9
Top:    804d1bd9b59c46e4fb16d3d7e275f8eb2b9a91f4
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2015-01-01 16:23:47 +0000

Temporary fix to avoid crashing during export

NB: GUI effects should be implemented properly eventually!


---

diff --git a/src/hid/gtk/gtkhid-gl.c b/src/hid/gtk/gtkhid-gl.c
index 1600c11..f29e8e9 100644
--- a/src/hid/gtk/gtkhid-gl.c
+++ b/src/hid/gtk/gtkhid-gl.c
@@ -1081,6 +1081,10 @@ ghid_screen_update (void)
 void
 ghid_set_lock_effects (hidGC gc, AnyObjectType *object)
 {
+  // XXX: Workaround to crashing exporters
+  if (gc->me_pointer != &ghid_hid)
+    return;
+
   /* Only apply effects to locked objects when in "lock" mode */
   if (Settings.Mode == LOCK_MODE &&
       TEST_FLAG (LOCKFLAG, object))
